### PR TITLE
Feature/refactor

### PR DESCRIPTION
**Pull request by doctorchebo**
I still have to work on the refactor, but this is what I have done so far:

Changes in Images class to import images in loop instead of one by one. 
Added a new Board class with the logic so main class is clean.
Created a UIElement Factory

### CODE REVIEW

**Comment by LuanaOrell** [comment]
@doctorchebo remove unnecessary files out/production

File: src/memory/Board.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+import javax.swing.border.BevelBorder;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class Board implements GameService{
```
Comment (LuanaOrell) [inline src/memory/Board.java]: >
```suggestion
public class Board implements GameService {
```

File: src/memory/Board.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    GameConstants gameConstants;
+
+    JFrame frame;
+    JPanel panelTitle, panelGrid, panelControl;
+    JButton new_game, solve, about;
```
Comment (LuanaOrell) [inline src/memory/Board.java:27]: avoid declaring multiple variable in a single line.

File: src/memory/Board.java [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+        about.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                JOptionPane.showMessageDialog(frame,"Just For Fun");
+            }
+        });
+    }
+
+    private void createLayout(){
```
Comment (LuanaOrell) [inline src/memory/Board.java]:
```suggestion
    private void createLayout() {
```

File: src/memory/Board.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        frame = (JFrame) uiFactory.getUIElement(UIType.FRAME).getElement("Memory");
```
Comment (LuanaOrell) [inline src/memory/Board.java]: Try to use strings into a constants

File: src/memory/FontEnlarger.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package memory;
+
+import java.awt.*;
```
Comment (LuanaOrell) [inline src/memory/FontEnlarger.java]: Avoid to use *, it is not a good practice

File: src/memory/Memory.java [hunk 6]
```diff
@@ -39,52 +39,23 @@
...
+    public Memory(Board board) {
     }
-    
-    private void enlargeFont(java.awt.Component c, float factor) {
-        c.setFont(c.getFont().deriveFont(c.getFont().getSize() * factor));
-    }
-    
-    public Memory(){ 
-        imagens = new Images();
+<<<<<<< HEAD
```
Comment (LuanaOrell) [inline src/memory/Memory.java]:
```suggestion
<<<<<<< HEAD
```
Resolve merge conflicts

File: src/memory/GameService.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package memory;
+
+public interface GameService {
```
Comment (LuanaOrell) [inline src/memory/GameService.java]: Please, try to create new packages in order to decouple the logic, e.g. a new package to UI components

**Review by LuanaOrell — CHANGES_REQUESTED** [review]

**Comment by alevap-ok** [comment]
Please review the merge conflicts so it is easier to review

File: src/memory/GameService.java (hunk 7 above)
Comment (doctorchebo) [reply src/memory/GameService.java]: done

File: src/memory/FontEnlarger.java (hunk 5 above)
Comment (doctorchebo) [reply src/memory/FontEnlarger.java]: done

File: src/memory/Board.java (hunk 4 above)
Comment (doctorchebo) [reply src/memory/Board.java]: done

File: src/memory/Board.java (hunk 2 above)
Comment (doctorchebo) [reply src/memory/Board.java:27]: done

File: src/memory/Board.java (hunk 1 above)
Comment (doctorchebo) [reply src/memory/Board.java]: done

**Pull request merged**